### PR TITLE
Rpf1 002 restaurant profile redux implementation

### PR DESCRIPTION
**Pull request by andress20**
Redux implementation in restaurantProfile.

Restaurant change from Class to Function. Implementation of hooks useEffect(GET and PUT request are now in reducer component), useHistory, useContext and useDispatch.

Restaurant form change in two input now are editables (name and email)

**modified files**
`RestProfile.js` input name and email are editable now
`RestaurantProfile.js` redux implementation

- [x] GET request now in reducer instead of component
- [x] PUT request now in reducer instead of component

restaurant

### CODE REVIEW

**Review by SergioJaramilloL — DISMISSED** [review]

File: src/pages/RestaurantProfile.js [hunk 1]
```diff
@@ -51,92 +65,72 @@ const MyOfficesAnchor = styled.a`
...
+    dispatch(getProfile())
+  },[])
+
+  const data = useSelector(
+    ({restaurantReducer: { 
+      ...state
+    }}) => {
+      return { ...state }
+    })
+  console.log(data.name)
```
Comment (simonhoyos) [inline src/pages/RestaurantProfile.js]: ojo con los console.logs

File: src/pages/RestaurantProfile.js [hunk 2]
```diff
@@ -208,24 +185,23 @@ class RestaurantProfile extends Component {
...
-            address={address}
-            phone={phone}
-            scheduleFrom={scheduleFrom}
-            scheduleTo={scheduleTo}
-            deposit={deposit}
-            nit={nit}
-            handleChange={this.handleChange}
-            handleSubmit={this.handleSubmit.bind(this)}
-            handleDeleteRestaurant={this.handleDeleteRestaurant.bind(this)}
+            //key={id}
```
Comment (simonhoyos) [inline src/pages/RestaurantProfile.js]: No dejar comentarios.

**Review by simonhoyos — DISMISSED** [review]

File: src/pages/RestaurantProfile.js [hunk 3]
```diff
@@ -167,34 +160,18 @@ class RestaurantProfile extends Component {
             });
             swal("Perfil eliminado exitosamente", "", "success");
             localStorage.removeItem('token');
-            this.context.isAuthenticated()
+            logout()
             this.props.history.push('/')
```
Comment (carosalazar28) [inline src/pages/RestaurantProfile.js]: Este a la final si se usa así? o con el hook de history

**Review by carosalazar28 — CHANGES_REQUESTED** [review]

File: src/pages/RestaurantProfile.js (hunk 3 above)
Comment (andress20) [reply src/pages/RestaurantProfile.js]: tienes razon ... es con el hook. Es que como no toque el handleDelete entonces tampoco me di cuenta, pero vamos a dejarlo entonces de una vez con el hook. gracias!

**Review by simonhoyos — DISMISSED** [review]

**Review by carosalazar28 — APPROVED** [review]

**Review by simonhoyos — APPROVED** [review]

**Review by SergioJaramilloL — APPROVED** [review]

**Pull request merged**